lib/service.js

```diff
@@ -360,13 +407,17 @@ ServicePrivate.prototype.listen = function() {
  */
 ServicePrivate.prototype.broadcast = function(type, data) {
   debug('broadcast', type, data);
+  this.postMessage('broadcast', {
+    type: type,
+    data: data
+  });
+};
+
+ServicePrivate.prototype.postMessage = function(messageId, data) {
```

Comment:
Can we get some jsdocs on all private function too? Even though they're not public it helps me follow the code :)